app/src/ui/delete-branch/index.tsx

```diff
@@ -39,8 +59,47 @@ export class DeleteBranch extends React.Component<IDeleteBranchProps, {}> {
     )
   }
 
-  private deleteBranch = () => {
-    this.props.dispatcher.deleteBranch(this.props.repository, this.props.branch)
-    this.props.dispatcher.closePopup()
+  private renderDeleteOnRemote() {
+    if (!!this.props.branch.remote) {
+      return (
+        <div>
+          <p>
+            <strong>
+              The branch also exists on the remote, do you wish to delete it
+              there as well?
+            </strong>
+          </p>
+          <Checkbox
+            label="Yes, delete this branch on the remote"
+            value={
+              this.state.includeRemoteBranch
+                ? CheckboxValue.On
+                : CheckboxValue.Off
+            }
+            onChange={this.onIncludeRemoteChanged}
+          />
+        </div>
+      )
+    }
+
+    return null
+  }
+
+  private onIncludeRemoteChanged = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const value = event.currentTarget.checked
+
+    this.setState({ includeRemoteBranch: value })
+  }
+
+  private deleteBranch = async () => {
+    await this.props.dispatcher.deleteBranch(
```

Comment:
I don't think we wanna `await` this, unless we're also gonna show the progress spinner, disable the form, etc.